File: 83_RemoveDuplicatesFromSortedList.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* deleteDuplicates(ListNode* head) {
        
        if(head == NULL)
        {
            return NULL;
        }
        map<int,int> mp;
        
        ListNode *temp = head;
        
        while(temp != NULL)
        {
            mp[temp->val] += 1;
            temp = temp->next;
        }
        
        ListNode *result = new ListNode();
        ListNode *headNode = result;
        
        for (const auto &item : mp) 
        {
            if(item.second > 0)
            {
                //ListNode *dummyObj = new ListNode(item.first);
                //result->next = dummyObj;
                //ListNode dummyObj(item.first);
                //cout << dummyObj.val << endl;
                //cout << item.first << endl;
                //ListNode *dummyObj = new ListNode(item.first);
                result->next = new ListNode(item.first);
                result = result->next;
            }
        }
        
        return headNode->next;
        //return head;
        
    }
};

File: 83_RemoveDuplicatesFromSortedList.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def deleteDuplicates(self, head: Optional[ListNode]) -> Optional[ListNode]:
        
        mp = {}
        
        temp = head
        
        while temp:
            if temp.val not in mp:
                mp[temp.val] = 1
            else:
                mp[temp.val] += 1
            temp = temp.next
        
        result = ListNode()
        newHead = result
        
        for key,element in mp.items():
            if element > 0:
                result.next = ListNode(key)
                result = result.next
        
        return newHead.next
                
        
        
        
        
#         numSet = set()
#         temp = head
        
#         while temp is not None:
#             numSet.add(head.val)
#             temp = temp.next
        
#         result = []
#         for element in numSet:
#             result.append(element)
#         return result
        